### DISCUSSION TITLE
o3d.t.geometry vs o3d.geometry namespaces

### DISCUSSION THREAD

**Discussion by marcelinorc** [isl-org/Open3D#6130] [Q&A]
Hello All, 

Sorry if this is too much of a noob question but, I'm terribly confused with the difference between the namespaces `o3d.t.geometry` and `o3d.geometry`.

Tutorials use both, and when trying to use results from one tutorial in another one, this gives a bunch of `TypeError: xxxx(): incompatible function arguments.`

To make things worst, the `t.geometry` namespace seems to have features that `geometry` doesn't and vice-versa. 

What is the difference between these namespaces and which one should I use?

Thanks a lot!
Marcelino

**Comment by cwreynolds** [comment]
@marcelinorc — did you intend to close this issue? I think you raise an excellent question.

I was just writing about this yesterday: https://github.com/isl-org/Open3D/issues/6060#issuecomment-1531976992 That discussion might be a tiny help in understanding this “legacy versus tensor” distinction, which I agree is both confusing and poorly documented.

I am personally only a month into using Open3D, but I hope I can eventually make contributions to the documentation so new users are not lead down the “legacy” path by the introductory doc and sample code.

**Reply by marcelinorc** [reply]
Yes, well, I found the reason for the 2 namespaces by looking at the code, so I closed the question. But if you think it is useful, I can reopen it.

**Reply by cwreynolds** [reply]
Thanks. I would like to think that a new user of Open3D could understand the basic architecture of the library without needing to first dive into the c++ and Python sources. 

I am collecting examples of this sort of confusion which I hope to eventually address by improving the doc/tutorials.

**Reply by saurabheights** [reply]
FYI - [6307](https://github.com/isl-org/Open3D/pull/6307)

I am working on documentation cleanup and once am done reviewing, I will add more docs. If you like, we can collaborate. P.S. Open3D Discord has documentation channel to further discuss.

**Reply by cwreynolds** [reply]
I just saw the email announcements. Thank you so much for this work @saurabheights. I have “Improve documentation” #6307 open in a tab and will look closer today.

**Comment by ssheorey** [comment]
Hi @cwreynolds hope to see your contribution to the docs / tutorials. We are in the process of adding GPU acceleration to Open3D, hence the existence of legacy and tensor APIs. We hope the conversion functions (to/from legacy) and different namespace (t) will help reduce confusion and make adoption easier. Most of the legacy API functionality is now available in the Tensor API (except for mesh processing functions), so we would encourage you to use the new tensor API as much as possible and take advantage of hardware acceleration.
